src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1603,112 +1646,98 @@ public RefreshItem(String keyInCache, AuthenticationRequest request, TokenCacheI
             mKey = keyInCache;
```

Comment:
I think besides key was used to get token, provide some enum value (Whether it's RT, MRRT or FRT) , then you don't need to generate cache key to compare